core/trino-main/src/main/java/io/trino/execution/RemoteTask.java

```diff
@@ -65,6 +65,8 @@
 
     void fail(Throwable cause);
 
+    void failWaitForUpdate(Throwable cause);
```

Comment:
nit: maybe think about a better name (though It's difficult for me to find a better one)

Reply:
Will rename to failRemotely